include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.cpp

```diff
@@ -306,6 +310,11 @@ SpatialOperator<dim, Number>::evaluate(BlockVectorType &       dst,
   // shift to the right-hand side of the equation
   dst *= -1.0;
 
+  if(param.right_hand_side)
+    rhs_operator.evaluate_scale_add(dst.block(block_index_pressure),
+                                    param.speed_of_sound * param.speed_of_sound,
+                                    time);
+
```

Comment:
From a design perspective, I am not sure whether we should do it like this. I first saw the definition of `evaluate_scale_add()` and expected the scaling factor to be a dynamic parameter, dynamic in the sense that it depends on the time step size for example (as we have for the mass operator). However, for the rhs-operator, which by definition operates on a `dealii::Function`, one might argue this scaling factor should appear inside `dealii::Function`.

Another reason is that the spatial derivative operators for acoustics are not written in `evaluate_scale()` logic, despite some terms are multiplied during operator evaluation by `c*c`. Hence, trying to understand the strong PDE model problem behind the code by just looking at the code (as I do right now), raises some questions/inconsistencies in my opinion.

Reply:
Would you prefer to add a scaling factor to `RHSKernelData`?

Reply:
This would have the advantage that we do not have to extend the interface of `RHSOperator` by a new function, and modify the existing interface of `RHSKernel`. In the current implementation, the kernel always multiplies by `scaling_factor` (even if we do not want to scale), so a `RHSKernelData::scaling_factor` initialized with `1.0` would probably be more straight forward.